src/client/datascience/jupyter/interactiveWindowDebugger.ts

```diff
@@ -213,8 +211,7 @@ export class InteractiveWindowDebugger implements IInteractiveWindowDebugger, IC
         // installed locally by the extension
         // Actually until this is resolved: https://github.com/microsoft/vscode-python/issues/7615, skip adding
         // this path.
-        const connectionInfo = kernel.connection;
-        if (connectionInfo && connectionInfo.localLaunch) {
+        if (isLocalConnection(kernel.kernelConnectionMetadata)) {
```

Comment:
This allows us to decouple the code form the `connection` information.
Getting the `connection` information is not cheap/easy.